### PR TITLE
E2e/playwright test improvements

### PR DESCRIPTION
**Pull request by lanej0**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

A number of improvements to Playwright tests to improve reliability.  Some re-factors based on feedback as well.

### How to test the changes in this Pull Request:

1. pnpm docker:up --filter=woocommerce
2. pnpx playwright test --config=e2e/playwright.config.js

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/e2e/tests/admin-analytics/analytics-overview.spec.js [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
+				'//button[@title="Choose which analytics to display and the section name"]'
+			);
+			await page.click( 'text=Move up' );
+			await page.click(
+				'//button[@title="Choose which analytics to display and the section name"]'
+			);
+			await page.click( 'text=Move up' );
+			await page.waitForTimeout( 1000 );
+		}
+	} );
```
Comment (lanej0) [inline plugins/woocommerce/e2e/tests/admin-analytics/analytics-overview.spec.js:41]: I moved all of the cleanup steps out to the afterEach hook.  That way they'll be run regardless of whether the test passes or fails -- seems to improve reliability.

File: plugins/woocommerce/e2e/tests/merchant/order-edit.spec.js [hunk 2]
```diff
@@ -175,6 +173,8 @@ test.describe( 'Edit order > Downloadable product permissions', () => {
 		await api.delete( `orders/${ noProductOrderId }`, { force: true } );
 	} );
 
+	// these tests aren't completely independent.  Needs some refactoring.
```
Comment (lanej0) [inline plugins/woocommerce/e2e/tests/merchant/order-edit.spec.js:176]: As the comment indicates, the following tests build on one another.  I think that might be the cause of them failing in subsequent runs without the environment being reset.

**Review by zhongruige — COMMENTED** [review]
Hi @lanej0! Thanks so much for getting this PR in! I was looking at this this morning and it doesn't seem like the pipeline ran against this PR for some reason. Maybe we just need to merge in `trunk`?

Screen Shot 2022-07-15 at 08 23 53

**Comment by lanej0** [comment]
@zhongruige -- that seems to have done it.  I rebased on trunk and now things are running.  Thanks for noticing that!

**Comment by Luc45** [comment]
It seems the Git History in this branch got messed up for some reason. Maybe it's worth creating another fresh branch from trunk and cherry-picking the relevant commits/code from this branch?

**Comment by lanej0** [comment]
Cleaned up over in https://github.com/woocommerce/woocommerce/pull/33965

**Pull request closed without merging**